lib/services/storageManagement2/lib/models/storageAccountUpdateParameters.js

```diff
@@ -40,107 +40,59 @@ var util = require('util');
-      if (this['tags'][valueElement1] !== null && this['tags'][valueElement1] !== undefined) {
-        if (typeof this['tags'][valueElement1].valueOf() !== 'string') {
-          throw new Error('this[\'tags\'][valueElement1] must be of type string.');
-        }
-        payload['tags'][valueElement1] = this['tags'][valueElement1];
-      }
-      else {
-        payload['tags'][valueElement1] = this['tags'][valueElement1];
-      }
-    }
-  }
-
-  if (this['accountType'] !== null && this['accountType'] !== undefined) {
-    var allowedValues = [ 'Standard_LRS', 'Standard_ZRS', 'Standard_GRS', 'Standard_RAGRS', 'Premium_LRS' ];
-    var thisaccountType = this['accountType'];
-    if (!allowedValues.some( function(item) { return item === thisaccountType; })) {
-      throw new Error(this['accountType'] + ' is not a valid value. The valid values are: ' + allowedValues);
-    }
-    if (payload['properties'] === null || payload['properties'] === undefined) {
-      payload['properties'] = {};
-    }
-    payload['properties']['accountType'] = this['accountType'];
-  }
-
-  if (this['customDomain']) {
-    if (payload['properties'] === null || payload['properties'] === undefined) {
-      payload['properties'] = {};
-    }
-    payload['properties']['customDomain'] = this['customDomain'].serialize();
-  }
-
-  return payload;
-};
-
-/**
- * Deserialize the instance to StorageAccountUpdateParameters schema
+ * Defines the metadata of StorageAccountUpdateParameters
  *
- * @param {JSON} instance
+ * @returns {object} metadata of StorageAccountUpdateParameters
  *
  */
-StorageAccountUpdateParameters.prototype.deserialize = function (instance) {
-  StorageAccountUpdateParameters['super_'].prototype.deserialize.call(this, instance);
-  if (instance) {
-    if (instance['tags']) {
-      this['tags'] = {};
-      for(var valueElement2 in instance['tags']) {
-        if (instance['tags'] !== null && instance['tags'] !== undefined) {
-          if (instance['tags'][valueElement2] !== undefined) {
-            this['tags'][valueElement2] = instance['tags'][valueElement2];
+StorageAccountUpdateParameters.prototype.mapper = function () {
+  return {
+    required: false,
+    serializedName: 'StorageAccountUpdateParameters',
+    type: {
+      name: 'Composite',
+      className: 'StorageAccountUpdateParameters',
+      modelProperties: {
+        tags: {
+          required: false,
+          serializedName: 'tags',
+          type: {
+            name: 'Dictionary',
+            value: {
+                required: false,
+                serializedName: 'StringElementType',
+                type: {
+                  name: 'String'
+                }
+            }
+          }
+        },
+        accountType: {
+          required: false,
+          serializedName: 'properties.accountType',
+          type: {
+            name: 'Enum',
+            allowedValues: [ 'Standard_LRS', 'Standard_ZRS', 'Standard_GRS', 'Standard_RAGRS', 'Premium_LRS' ]
```

Comment:
Premium_LRS and Standard_ZRS accounts cannot be changed to other account types, and other account types cannot be changed to Standard_ZRS or Premium_LRS. So these two should not in the allowed values set.